plugins/woocommerce-blocks/assets/js/blocks/checkout/inner-blocks/checkout-contact-information-block/block.tsx

```diff
@@ -36,47 +41,45 @@ const Block = (): JSX.Element => {
 		dispatchCheckoutEvent( 'set-email-address' );
 	};
 
-	const createAccountUI = ! customerId &&
+	const createAccountVisible =
+		! customerId &&
 		getSetting( 'checkoutAllowsGuest', false ) &&
-		getSetting( 'checkoutAllowsSignup', false ) && (
-			<CheckboxControl
-				className="wc-block-checkout__create-account"
-				label={ __( 'Create an account?', 'woocommerce' ) }
-				checked={ shouldCreateAccount }
-				onChange={ ( value ) =>
-					__internalSetShouldCreateAccount( value )
-				}
-			/>
-		);
+		getSetting( 'checkoutAllowsSignup', false );
+
+	const createAccountUI = createAccountVisible && (
+		<CheckboxControl
+			className="wc-block-checkout__create-account"
+			label={ __( 'Create an account?', 'woocommerce' ) }
+			checked={ shouldCreateAccount }
+			onChange={ ( value ) => __internalSetShouldCreateAccount( value ) }
+		/>
+	);
+
+	const onChangeForm = ( newAddress ) => {
```

Comment:
This is a TS error - `newAddress` needs a type,I think this works?

`newAddress: { email: string } & AdditionalValues`

Reply:
Ah, but then this causes a problem with `onChange` on the `Form` down below 😟

Reply:
Wondering if we need to make `Form` take a generic param which allows us to define the expected args for `onChange`

Reply:
I tried that and went into a rabbit hole, happy to try it again.

Reply:
Maybe for now we can leave it as `any` and add a `@ts-ignore` a follow up issue could be to review and fix this